### PR TITLE
DOC: Fix dsintro.rst flake8 issues

### LINKED ISSUE #24175: DOC: Fix flake8 issues in doc/source/dsintro.rst
We didn't start validating the format of PEP8 and other code standards in the documentation examples until recently. We still have some files with errors, that we need to skip, and that we should fix, so we can start validating them.

The first step of this issue would be edit `setup.cfg` in the pandas home, and in the `flake8-rst` section, remove from the `exclude` list the file `doc/source/dsintro.rst`

After that, running the next command will report the errors in the file (note that syntax error usually prevent to validate other errors, and the list of errors to fix can become much longer when the syntax error is fixed (please make sure that you are using `flake8-rst` version `0.7.0` or higher):

```
$ flake8-rst doc/source/dsintro.rst 
doc/source/dsintro.rst:74:12: E203 whitespace before ':'
doc/source/dsintro.rst:74:21: E203 whitespace before ':'
doc/source/dsintro.rst:74:30: E203 whitespace before ':'
doc/source/dsintro.rst:95:12: E203 whitespace before ':'
doc/source/dsintro.rst:95:22: E203 whitespace before ':'
doc/source/dsintro.rst:95:32: E203 whitespace before ':'
doc/source/dsintro.rst:307:15: E203 whitespace before ':'
doc/source/dsintro.rst:308:15: E203 whitespace before ':'
doc/source/dsintro.rst:337:14: E203 whitespace before ':'
doc/source/dsintro.rst:338:14: E203 whitespace before ':'
doc/source/dsintro.rst:349:44: E231 missing whitespace after ','
doc/source/dsintro.rst:349:56: E231 missing whitespace after ','
doc/source/dsintro.rst:350:17: E231 missing whitespace after ','
doc/source/dsintro.rst:350:20: E231 missing whitespace after ','
doc/source/dsintro.rst:350:33: E231 missing whitespace after ','
doc/source/dsintro.rst:350:36: E231 missing whitespace after ','
doc/source/dsintro.rst:511:28: E251 unexpected spaces around keyword / parameter equals
doc/source/dsintro.rst:511:30: E251 unexpected spaces around keyword / parameter equals
doc/source/dsintro.rst:519:27: E251 unexpected spaces around keyword / parameter equals
doc/source/dsintro.rst:519:29: E251 unexpected spaces around keyword / parameter equals
doc/source/dsintro.rst:519:41: W504 line break after binary operator
doc/source/dsintro.rst:535:27: E251 unexpected spaces around keyword / parameter equals
doc/source/dsintro.rst:535:29: E251 unexpected spaces around keyword / parameter equals
doc/source/dsintro.rst:536:27: E251 unexpected spaces around keyword / parameter equals
doc/source/dsintro.rst:536:29: E251 unexpected spaces around keyword / parameter equals
doc/source/dsintro.rst:708:27: E203 whitespace before ':'
doc/source/dsintro.rst:708:44: E203 whitespace before ':'
doc/source/dsintro.rst:708:56: E202 whitespace before '}'
doc/source/dsintro.rst:709:27: E203 whitespace before ':'
doc/source/dsintro.rst:709:44: E203 whitespace before ':'
doc/source/dsintro.rst:709:56: E202 whitespace before '}'
doc/source/dsintro.rst:749:30: E231 missing whitespace after ','
doc/source/dsintro.rst:780:13: W605 invalid escape sequence '\.'
doc/source/dsintro.rst:801:38: E261 at least two spaces before inline comment
doc/source/dsintro.rst:809:12: E225 missing whitespace around operator
doc/source/dsintro.rst:809:40: E231 missing whitespace after ','
doc/source/dsintro.rst:813:40: E231 missing whitespace after ','
doc/source/dsintro.rst:816:40: E231 missing whitespace after ','
doc/source/dsintro.rst:836:29: E203 whitespace before ':'
doc/source/dsintro.rst:837:29: E203 whitespace before ':'
doc/source/dsintro.rst:846:18: E225 missing whitespace around operator
doc/source/dsintro.rst:846:22: E225 missing whitespace around operator
doc/source/dsintro.rst:846:22: E999 SyntaxError: invalid syntax
doc/source/dsintro.rst:893:19: E203 whitespace before ':'
doc/source/dsintro.rst:894:19: E203 whitespace before ':'
doc/source/dsintro.rst:950:53: E231 missing whitespace after ','
doc/source/dsintro.rst:950:70: E231 missing whitespace after ','
doc/source/dsintro.rst:950:72: E231 missing whitespace after ','
doc/source/dsintro.rst:950:74: E231 missing whitespace after ','
doc/source/dsintro.rst:950:77: E231 missing whitespace after ','
doc/source/dsintro.rst:950:80: E231 missing whitespace after ','
doc/source/dsintro.rst:950:82: E231 missing whitespace after ','
doc/source/dsintro.rst:950:83: E501 line too long (85 > 79 characters)
doc/source/dsintro.rst:950:84: E231 missing whitespace after ','
doc/source/dsintro.rst:951:26: E203 whitespace before ':'
```

Once all the errors are addressed, please open a pull request with the fixes in the file, and removing the file from `setup.cfg`. If you need to do something that feels wrong to fix an error, please ask in a comment to this issue. Please avoid other unrelated changes, which can be addressed in a separate pull request.

### PR DESCRIPTION
**Pull request by addisonlynch**
- [x] closes #24175 
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

File: doc/source/dsintro.rst [hunk 1]
```diff
@@ -777,7 +775,7 @@ R package):
    :okwarning:
 
    # restore GlobalPrintConfig
-   pd.reset_option('^display\.')
+   pd.reset_option('^display\.')  # noqa: W605
```
Comment (datapythonista) [inline doc/source/dsintro.rst]:
```suggestion
   pd.reset_option(r'^display\.')
```

**Review by datapythonista — COMMENTED** [review]
Thanks @addisonlynch 

If you can just fix the commented error instead of adding the `noqa`.

Also, if you can add the issue number and check the checkbox, so the issue is closed when we merge this.

File: doc/source/dsintro.rst (hunk 1 above)
Comment (addisonlynch) [reply doc/source/dsintro.rst]: Done

**Review by datapythonista — APPROVED** [review]
lgtm, thanks @addisonlynch

**Comment by jreback** [comment]
thanks @addisonlynch

**Pull request merged**